File: playground/java/src/main/java/com/algolia/playground/AgentStudio.java
package com.algolia.playground;

import com.algolia.api.AgentStudioClient;
import io.github.cdimascio.dotenv.Dotenv;

public class AgentStudio {

  public static void main(String[] args) throws Exception {
    var dotenv = Dotenv.configure().directory("../").load();
    var appId = dotenv.get("ALGOLIA_APPLICATION_ID");
    var apiKey = dotenv.get("ALGOLIA_ADMIN_KEY");

    var client = new AgentStudioClient(appId, apiKey);

    try {
      var response = client.listAgents();
      System.out.println("List of agents:");
      for (var agent : response.getData()) {
        System.out.println("- " + agent.getName() + " (ID: " + agent.getId() + ")");
      }
    } finally {
      client.close();
    }
  }
}

File: playground/kotlin/src/main/kotlin/com/algolia/playground/AgentStudio.kt
package com.algolia.playground

import com.algolia.client.api.AgentStudioClient
import io.github.cdimascio.dotenv.Dotenv
import kotlin.system.exitProcess

suspend fun main() {
    val dotenv = Dotenv.configure().directory("../").load()

    val client = AgentStudioClient(
        appId = dotenv["ALGOLIA_APPLICATION_ID"],
        apiKey = dotenv["ALGOLIA_ADMIN_KEY"],
    )

    val response = client.listAgents()
    println("List of agents:")
    for (agent in response.agents) {
        println("- ${agent.name} (ID: ${agent.id})")
    }

    exitProcess(0)
}
